app/player/search/more/page.tsx

```diff
@@ -0,0 +1,182 @@
+"use client";
+
+import { useContext } from "react";
+import { SpotifyApiContext } from "../../context";
+import { useRouter, useSearchParams } from "next/navigation";
+import invariant from "tiny-invariant";
+import { SpotifyAPi } from "../../../../components/Spotify/SpotifyApi";
+import styled from "styled-components";
+import {
+  Album,
+  Artist,
+  SearchResponse,
+  SpotifyTrackItem,
+} from "../../../../components/types";
+import { BorderlessButton } from "../../../../components/IconButtons/IconButtons";
+import { PlayableItem } from "../../../../components/PlayableItem";
+import { LoadMoreContext, SearchResultContext } from "../searchContext";
+import { PlayPauseButton } from "../../../../components/Buttons/PlayPauseButton";
+import { Spinner } from "../../../../components/Spinner";
+
+export default () => {
+  const router = useRouter();
+  const searchParams = useSearchParams();
+  const searchQuery = searchParams?.get("searchQuery");
+  const itemType = searchParams?.get("itemType");
+  const { spotifyApiRef } = useContext<{ spotifyApiRef: SpotifyAPi }>(
+    SpotifyApiContext
+  );
+
+  invariant(searchQuery, "search query is required");
+  invariant(itemType, "itemType is required");
+
+  if (itemType != "artist" && itemType != "album" && itemType != "track") {
+    throw new Error(`Item type ${itemType} not supported yet`);
+  }
+
+  const { data } = useContext<{ data: SearchResponse | null }>(
+    SearchResultContext
+  );
+
+  const { loadMore, canFetchMore, isFetching } = useContext(LoadMoreContext);
+
+  if (!data) {
+    return null;
+  }
+
+  let items: Album[] | Artist[] | SpotifyTrackItem[] = [];
+
+  if (itemType == "album" && data.albums) {
+    items = data.albums.items;
+  }
+
+  if (itemType == "artist" && data.artists) {
+    items = data.artists.items;
+  }
+
+  if (itemType == "track" && data.tracks) {
+    items = data.tracks.items;
+  }
+
+  const openDetailsPage = (itemId: string) => {
+    switch (itemType) {
+      case "album":
+      case "track": {
+        const queryParams = new URLSearchParams({
+          albumId: itemId,
+        });
+        router.push("/player/album?" + queryParams.toString());
+        return;
+      }
+
+      case "artist": {
+        const queryParams = new URLSearchParams({
+          artistId: itemId,
+        });
+        router.push("/player/artist?" + queryParams.toString());
+        return;
+      }
+
+      default: {
+        throw new Error("unknown item type");
+      }
+    }
+  };
+
+  const playItem = (uri: string) => {
+    spotifyApiRef.playPlayback({ context_uri: uri });
+  };
+
+  const titleTexts = {
+    album: "Albums",
+    artist: "Artists",
+    track: "Tracks",
+  };
+
+  return (
+    <Container>
+      <h1>{titleTexts[itemType]}</h1>
+      <ItemContainer>
+        {items.map((item: Album | Artist | SpotifyTrackItem) => {
```

Comment:
could we simplify this by extracting the map function out of here and having a switch-case there?